ggshield/cmd/secret/scan/secret_scan_common_options.py

```diff
@@ -139,6 +139,17 @@ def _banlist_detectors_callback(
 )
 
 
+_all_secrets = click.option(
+    "--all-secrets",
+    is_flag=True,
```

Comment:
You need to add a `default=None` otherwise the option won't work when it appears early on the command line (`secret scan path --all-secrets .` is OK, but `secret --all-secrets scan path .` is not).